src/dashboards/components/DashboardContainer.tsx

```diff
@@ -21,17 +21,92 @@ import {AUTOREFRESH_DEFAULT} from 'src/shared/constants'
 // Types
 import {AppState, ResourceType, AutoRefreshStatus} from 'src/types'
 
+// Notifications
+import {dashboardAutoRefreshTimeoutSuccess} from 'src/shared/copy/notifications'
+import {notify} from 'src/shared/actions/notifications'
+
 const {Active} = AutoRefreshStatus
 
 type ReduxProps = ConnectedProps<typeof connector>
 type Props = ReduxProps
 
+let timer
```

Comment:
Because using state introduces infinite rerenders. declaring the timer variable from outside the component avoids this entirely. cc @drdelambre

Reply:
This was addressed. Resolving.

Reply:
>Because using state introduces infinite rerenders.

this seems like a smell. what is the underlying issue that is causing re-renders?

Reply:
Hard to ascertain, discussed w/ @drdelambre and it seems like its because the state update loses reference to the right settimeout, something which does not happen if the variable for timer is either in the ref or outside of the component. Per @drdelambre 's suggestion, the ref solution was implemented